skeleton/networkxGraphFromArray.py

```diff
@@ -0,0 +1,151 @@
+(-1 -1 1)  (-1 0 1)  (-1 1 1)
+
+(0 -1 -1) (0 0 -1) (0 1 -1)
+(0 -1 0)  (0 0 0)  (0 1 0)
+(0 -1 1)  (0 0 1)  (0 1 1)
+
+(1 -1 -1) (1 0 -1) (1 1 -1)
+(1 -1 0)  (1 0 0)  (1 1 0)
+(1 -1 1)  (1 0 1)  (1 1 1)
+"""
+
+# permutations of (-1, 0, 1) in three/two dimensional tuple format
+# representing 8 and 26 increments around a pixel at origin (0, 0, 0)
+# 2nd ordered neighborhood around a voxel/pixel
+LISTSTEPDIRECTIONS3D = list(itertools.product((-1, 0, 1), repeat=3))
+LISTSTEPDIRECTIONS3D.remove((0, 0, 0))
+
+LISTSTEPDIRECTIONS2D = list(itertools.product((-1, 0, 1), repeat=2))
+LISTSTEPDIRECTIONS2D.remove((0, 0))
+
+
+def _getIncrements(configNumber, dimensions):
+    """
+    Return position of non zero voxels/pixels in the
+    binary string of config number
+    Parameters
+    ----------
+    configNumber : int64
+        integer less than 2 ** 26
+
+    dimensions: int
+        number of dimensions, can only be 2 or 3
+
+    Returns
+    -------
+    list
+        a list of incremental direction of a non zero voxel/pixel
+
+    Notes
+    ------
+    As in the beginning of the program, there are incremental directions
+    around a voxel at origin (0, 0, 0) which are returned by this function.
+    configNumber is a decimal number representation of 26 binary numbers
+    around a voxel at the origin in a second ordered neighborhood
+    """
+    configNumber = np.int64(configNumber)
+    if dimensions == 3:
+        # convert decimal number to a binary string
+        neighborValues = [(configNumber >> digit) & 0x01 for digit in range(26)]
+        return [neighborValue * increment for neighborValue, increment in zip(neighborValues, LISTSTEPDIRECTIONS3D)]
+    else:
+        neighborValues = [(configNumber >> digit) & 0x01 for digit in range(8)]
+        return [neighborValue * increment for neighborValue, increment in zip(neighborValues, LISTSTEPDIRECTIONS2D)]
+
+
+def _setAdjacencyList(arr):
+    """
+    Return position of non zero voxels/pixels in the
+    binary string of config number
+    Parameters
+    ----------
+    arr : numpy array
+        binary numpy array can only be 2D Or 3D
+
+    Returns
+    -------
+    dictOfIndicesAndAdjacentcoordinates: Dictionary
+        key is the nonzero coordinate in input "arr" and value
+        is all the position of nonzero coordinates around it
+        in it's second order neighborhood
+
+    """
+    dimensions = arr.ndim
+    if dimensions == 3:
+        # flipped 3D template in advance
+        template = np.array([[[33554432, 16777216, 8388608], [4194304, 2097152, 1048576], [524288, 262144, 131072]],
+                            [[65536, 32768, 16384], [8192, 0, 4096], [2048, 1024, 512]],
+                            [[256, 128, 64], [32, 16, 8], [4, 2, 1]]], dtype=np.uint64)
+    elif dimensions == 2:
+        # 2 dimensions
+        template = np.array([[2 ** 0, 2 ** 1, 2 ** 2], [2 ** 3, 0, 2 ** 4], [2 ** 5, 2 ** 6, 2 ** 7]])
+        template = np.fliplr(np.flipud(template))
```

Comment:
Why do you flip template twice? Can't you create it with the correct values in the first line?